### ISSUE TITLE
Support login flow in browser (v2)

### ISSUE THREAD

**Issue by tobiasKaminsky** [nextcloud/preferred_providers#24]
As briefly discussed with @skjnldsv using login flow v2 is not possible, as the Desktop client does not know to which provider it should connect.
So we need a "v3" with a gateway
- Desktop starts simple sign up in browser with generated token
  - pings nc.com/signup for continous monitoring (same like v2)
- User chooses provider on web site, creates account, etc.
- after successful creation of the account the desired provider send nc:// info to Desktop via /signup (identified via token)

Motivation: 
- QT for Apple M1 currently does not support QT Webkit
- there might be proxy settings where it is better to use standard browser for this signup/login
- getting rid of v1 code on Desktop, as it is only used for preferred providers

cc @FlexW

**Comment by tobiasKaminsky** [comment]
No idea on server side.
Desktop should be very easy, as we only need to ping/poll another side and add parsing of server url.

**Comment by AndyScherzinger** [comment]
@tobiasKaminsky @skjnldsv I put it on the planning column via a distinct server ticket, feel free to move that one if you want it in a different repo also at Tobias, can you create the corresponding client ticket and link it here please? 🙏

**Comment by tobiasKaminsky** [comment]
Server: https://github.com/nextcloud/server/issues/29542
Desktop: https://github.com/nextcloud/desktop/issues/3953

**Comment by tobiasKaminsky** [comment]
This is also the case for GSS.

As WebView (v1) causes nowadays some trouble on Android it would be great to start with this foundation work soon, so that all clients can migrate.

**Comment by skjnldsv** [comment]
Can you share the specs and documentation of the v3 ?

**Comment by tobiasKaminsky** [comment]
Not really, as this is "only" about what we need, not how we can do it.
From my/client pov it would be:
- client asks https://cloud.example.com/index.php/login/v3 or  nc.com/signup/login (for preferred providers)
- poll endpoint will be returned
- user completes login (same like v2)
- meanwhile the endpoint needs to 
  - either update polling endpoint to e.g. provider1 or 
  - provider1 needs to pass credentials to old endpoint
(this is where I am not sure how to do it, as I am not an expert on server)
- endpoint then returns  200 with json object containing all information.

So basically it is same as v2, but it must be possible to get redirect from first server (simple signup or GSS entry node) to real server.

Flow v2: https://docs.nextcloud.com/server/latest/developer_manual/client_apis/LoginFlow/index.html#login-flow-v2

**Comment by skjnldsv** [comment]
I fail to see how the login flow is relevant here?
It's only about what we return as an `nc:` url, right?
https://github.com/nextcloud/preferred_providers/blob/9d07dd6775a315c77b7ac0bdff0d80f31a12dde8/lib/Controller/PasswordController.php#L301

**Comment by tobiasKaminsky** [comment]
On simple signup, client currently use webview, where all is handled within it.
In the end then the redirect url is changed, that's true.

But this cannot work for flow v2, and thus we need v3.

Imagine v2 with http://www.nextcloud.com/register
-> this then needs to return a poll url, that is already on target server, or return a generic url, that then returns correct nc:// string.

**Comment by skjnldsv** [comment]
But the preferred providers never used the login flow ?
We generate our own app token, and just inform the client of the url/userid/pass data.

**Comment by Rello** [comment]
Hi,

I am picking up on this 

```
The matching handler is in webview.cpp (line 81): it registers the nc WebEngine scheme and installs a handler. That handler parses URLs like:

nc://login/server:<server>&user:<user>&password:<password>
in webview.cpp (line 173), extracting server:, user:, and password: from the path and emitting them to the setup wizard.

However, the OS-level nc:// handler is different. The application-level argument handling only accepts nc://open... for local file editing in application.cpp (line 934). So if an external browser launches the installed client with nc://login/..., current app-level dispatch will not route it into account setup. The modern login path is Login Flow v2 via /index.php/login/v2 and polling, not this redirect URI.
```

The idea is now that we handle the URL Scheme at the application level - and we change this app to **not add user/pw when coming from the desktop client**.  Reason is, that any other app could steal the credentials

**Why would this still work**
So the user does the Process in the normal Webbrowser. when this is finished, `nc://login/server:<server>` will init the current V2 flow, but jump directly to the "grand access" page. There the V2 polling is done as now. 
So its just 2 extra clicks in this case.

<img width="1495" height="538" alt="Image" src="https://github.com/user-attachments/assets/35756048-c7d7-42cf-a639-287d1eaf5337" />

**Why?**
We urgently need to get rid of the QTWebEngine in the desktop client. it is not working anymore due to sandboxing.

**Data:**
we are talking about 700 weekly potential users vs. 20k from the website. So the impact of the extra clicks is limited

**Comment by skjnldsv** [comment]
Done
